### PR TITLE
Update Filing Service for Relevant Period new Statements

### PR DESCRIPTION
**Pull request by amalviyach**
### JIRA link

https://companieshouse.atlassian.net/browse/ROECCT-143

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by markpit** [comment]
The 2 code smells on the sonar report will need fixing

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java [hunk 1]
```diff
@@ -28,6 +29,9 @@ public class UpdateSubmission {
     public static final String ADDITIONS_FIELD = "additions";
     public static final String CESSATIONS_FIELD = "cessations";
     public static final String TRUSTS_ADDITIONS_FIELD = "trusts";
+    public static final String CHANGE_BO_RELEVANT_PERIOD = "changeBORelevantPeriod";
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:32]: these constant names should end _FIELD to match the other field names

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java [hunk 2]
```diff
@@ -53,6 +57,15 @@ public class UpdateSubmission {
     @JsonProperty(BENEFICIAL_OWNERS_FIELD)
     private BeneficialOwnersStatementType beneficialOwnerStatement;
 
+    @JsonProperty(CHANGE_BO_RELEVANT_PERIOD)
+    private RelevantStatementsType changeBORelevantPeriod;
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:61]: i think the class attributes should have the word Statement in them as a period can be different from a statement
eg changeBORelevantPeriodStatement
trusteeInvolvedRelevantPeriodStatement
changeBeneficiaryRelevantPeriodStatement

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java [hunk 3]
```diff
@@ -53,6 +57,15 @@ public class UpdateSubmission {
...
+
+    @JsonProperty(TRUSTEE_INVOLVED_RELEVANT_PERIOD)
+    private RelevantStatementsType trusteeInvolvedRelevantPeriod;
+
+    @JsonProperty(CHANGE_BENEFICIARY_RELEVANT_PERIOD)
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:66]: changes to the model classes will need to be added to the /spec/schema.json file as well

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:66]: ignore this, i looked and saw that they have been previously added

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java [hunk 4]
```diff
@@ -28,6 +29,9 @@ public class UpdateSubmission {
     public static final String ADDITIONS_FIELD = "additions";
     public static final String CESSATIONS_FIELD = "cessations";
     public static final String TRUSTS_ADDITIONS_FIELD = "trusts";
+    public static final String CHANGE_BO_RELEVANT_PERIOD = "changeBORelevantPeriod";
+    public static final String TRUSTEE_INVOLVED_RELEVANT_PERIOD = "trusteeInvolvedRelevantPeriod";
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java:33]: Just wondering if this constant and the associated field should also be prefixed with 'change'?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java (hunk 1 above)
Comment (amalviyach) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java]: done

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java (hunk 4 above)
Comment (amalviyach) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/UpdateSubmission.java]: checked it should be left like this

**Review by markpit — APPROVED** [review]

**Review by mwestacott — APPROVED** [review]

**Pull request merged**